builder/dockerfile/internals_windows.go

```diff
@@ -0,0 +1,392 @@
+
+	err := windows.RegQueryInfoKey(userNamesKey, nil, nil, nil, &subKeyCount, &maxSubKeyLen, nil, nil, nil, nil, nil, nil)
+	if err != nil {
+		return "", false, err
+	}
+
+	accountLocated := false
+	accountBuffer := make([]uint16, maxSubKeyLen+1)
+	maximumAccountBufferLength := uint32(len(accountBuffer))
+	accountBufferLength := maximumAccountBufferLength
+	accountSid := ""
+
+	for index := uint32(0); index < subKeyCount; index++ {
+		accountBufferLength = maximumAccountBufferLength
+		err := windows.RegEnumKeyEx(userNamesKey, index, &accountBuffer[0], &accountBufferLength, nil, nil, nil, nil)
+
+		if err != nil {
+			return "", false, err
+		}
+
+		accountName = windows.UTF16ToString(accountBuffer[:])
+
+		err = changeKeySecurity(userNamesKey, accountName)
+		if err != nil {
+			return "", false, err
+		}
+
+		if accountName == userStr {
+			accountLocated = true
+			break
+		}
+	}
+
+	if accountLocated {
+		var accountKey windows.Handle
+
+		err := windows.RegOpenKeyEx(userNamesKey, &accountBuffer[0], 0, windows.KEY_READ, &accountKey)
+		if err != nil {
+			return "", false, err
+		}
+
+		var userRid uint32
+
+		err = windows.RegQueryValueEx(accountKey, nil, nil, &userRid, nil, nil)
+		if err != nil {
+			return "", false, err
+		}
+
+		accountSid = computerSid + "-" + strconv.FormatUint(uint64(userRid), 10)
+
+		defer windows.RegCloseKey(accountKey)
+	}
+
+	return accountSid, accountLocated, nil
+}
+
+func lookupNTGroup(groupNamesKey windows.Handle, groupStr string, computerSid string) (string, bool, error) {
+
+	var subKeyCount uint32
+	var maxSubKeyLen uint32
+	var accountName string
+
+	err := windows.RegQueryInfoKey(groupNamesKey, nil, nil, nil, &subKeyCount, &maxSubKeyLen, nil, nil, nil, nil, nil, nil)
+	if err != nil {
+		return "", false, err
+	}
+
+	accountLocated := false
+	accountBuffer := make([]uint16, maxSubKeyLen+1)
+	maximumAccountBufferLength := uint32(len(accountBuffer))
+	accountBufferLength := maximumAccountBufferLength
+	accountSid := ""
+
+	for index := uint32(0); index < subKeyCount; index++ {
+		accountBufferLength = maximumAccountBufferLength
+		err := windows.RegEnumKeyEx(groupNamesKey, index, &accountBuffer[0], &accountBufferLength, nil, nil, nil, nil)
+
+		if err != nil {
+			return "", false, err
+		}
+
+		accountName = windows.UTF16ToString(accountBuffer[:])
+
+		err = changeKeySecurity(groupNamesKey, accountName)
+		if err != nil {
+			return "", false, err
+		}
+
+		if accountName == groupStr {
+			accountLocated = true
+			break
+		}
+	}
+
+	if accountLocated {
+		var accountKey windows.Handle
+
+		err := windows.RegOpenKeyEx(groupNamesKey, &accountBuffer[0], 0, windows.KEY_READ, &accountKey)
+		if err != nil {
+			return "", false, err
+		}
+
+		var groupRid uint32
+
+		err = windows.RegQueryValueEx(accountKey, nil, nil, &groupRid, nil, nil)
+		if err != nil {
+			return "", false, err
+		}
+
+		accountSid = computerSid + "-" + strconv.FormatUint(uint64(groupRid), 10)
+
+		defer windows.RegCloseKey(accountKey)
```

Comment:
As with `logonNTUser`, this defer should be moved up, right after opening the key.